src/e2e-test/java/io/cdap/cdap/ui/stepsdesign/SnackBar.java

```diff
@@ -16,58 +16,24 @@
 
 package io.cdap.cdap.ui.stepsdesign;
 
-import io.cdap.cdap.ui.utils.Constants;
 import io.cdap.cdap.ui.utils.Helper;
 import io.cdap.e2e.utils.ElementHelper;
 import io.cdap.e2e.utils.SeleniumDriver;
 import io.cdap.e2e.utils.WaitHelper;
-import io.cucumber.java.en.Given;
 import io.cucumber.java.en.Then;
 import org.junit.Assert;
 import org.openqa.selenium.By;
 import org.openqa.selenium.WebElement;
 
-import java.util.ArrayList;
-import java.util.List;
 
 public class SnackBar {
-  @Given("Navigate to Home Page")
-  public void navigateToHomePage() {
-    try {
-      SeleniumDriver.openPage(Constants.WRANGLE_HOME_URL);
-      WaitHelper.waitForPageToLoad();
-    } catch (Exception e) {
-      System.err.println("error:" + e);
-    }
-  }
-
-  @Then("Click on the data exploration card")
-  public void clickOnTheDataExplorationCard() {
-    try {
-      WaitHelper.waitForPageToLoad();
-      List<String> productName = new ArrayList<String>();
-      List<WebElement> allProductsName = SeleniumDriver.getDriver().findElements(
-          By.xpath(".//*[@data-testid='wrangler-home-ongoing-data-exploration-card']"));
-      ElementHelper.clickOnElement(allProductsName.get(0));
-      String url = SeleniumDriver.getDriver().getCurrentUrl();
-      Assert.assertTrue(url.contains("http://localhost:11011/cdap/ns/default/wrangler-grid"));
-    } catch (Exception e) {
-      System.err.println("error:" + e);
-    }
-  }
-
   @Then("Click on the Snackbar close icon")
   public void verifyTheSnackbarPopUpIsComingOrNot() {
-    try {
-      WaitHelper.waitForPageToLoad();
-      WaitHelper.waitForElementToBeDisplayed(Helper.locateElementByTestId("snackbar-alert"));
-      Assert.assertTrue(ElementHelper.isElementDisplayed(Helper.locateElementByTestId("snackbar-alert")));
-      WaitHelper.waitForElementToBeDisplayed(Helper.locateElementByTestId("snackbar-close-icon"));
-      WaitHelper.waitForElementToBeClickable(Helper.locateElementByTestId("snackbar-close-icon"));
-      WebElement ele = SeleniumDriver.getDriver().findElement(By.xpath("//*[@data-testid = 'snackbar-close-icon']"));
-      ElementHelper.clickOnElement(ele);
-    } catch (Exception e) {
-      System.err.println("error:" + e);
-    }
+    WaitHelper.waitForElementToBeDisplayed(Helper.locateElementByTestId("snackbar-alert"));
+    Assert.assertTrue(ElementHelper.isElementDisplayed(Helper.locateElementByTestId("snackbar-alert")));
+    WaitHelper.waitForElementToBeDisplayed(Helper.locateElementByTestId("snackbar-close-icon"));
+    WaitHelper.waitForElementToBeClickable(Helper.locateElementByTestId("snackbar-close-icon"));
+    WebElement ele = Helper.locateElementByTestId("snackbar-close-icon");
```

Comment:
I feel like there's a misunderstanding here. I see these wait statements across all classes. You don't have to wait for every element to be displayed/clickable before attempting a click on it or check for its existense. These wait statements only make sense when there's an actual lag/loading happening (e.g., page load, api call). Most of the time, you directly try to click, and if it fails, it means something's wrong regardless if you waited or not. Lines 34 and 35 seem redundant to me.

Reply:
Please remove these `WaitHelper.waitForElementToBeDisplayed()`, `WaitHelper.waitForElementToBeClickable()` unless waiting is really needed like I said in the above comment.

Reply:
sure @mrahanjam will remove that wait statements

Reply:
we have removed `WaitHelper.waitForElementToBeClickable`
and we have kept `waitForElementToBeDisplayed ` as we need it here